doc/source/whatsnew/v1.0.0.rst

```diff
@@ -298,45 +298,6 @@ New repr for :class:`~pandas.arrays.IntervalArray`
    pd.arrays.IntervalArray.from_tuples([(0, 1), (2, 3)])
 
 
-All :class:`SeriesGroupBy` aggregation methods now respect the ``observed`` keyword
```

Comment:
Two things

1. I think this is a bugfix, not an API change
2. I don't think it warrants an entire subsection